required/latex-lab/latex-lab-math.dtx

```diff
@@ -2083,9 +2085,82 @@
         \tag_socket_use:n {math/display/begin}
         \tag_socket_use:nn{math/display/formula/begin}{}{#1}
       }
+    \@@_prepare_display_end:
     $$
   }
 %    \end{macrocode}
+%    
+%    \begin{macrocode}
+\cs_new_protected:Npn \@@_prepare_display_end: {  
+%    \end{macrocode}
+%    Flipping the \cs{belowdisplay} values is done so that we get
+%    (assumption) a negative skip and not make the page bigger.  Then
+%    in \cs{@@_tag_dollardollar_display_end:} we take that out again,
+%    add the tagging code and then we put a real
+%    \cs{postdisplaypenalty} in and the right skip. This might need
+%    some refinement if that skip is actually negative from the start
+%    (not sure it ever is and is worth bothering about).
+% \changes{v0.6n}{2025/03/07}{Flipping the sign of the ``below skips'' at
+%    when the formula ends and not in \cs{everydisplay} (tagging/809)}
+%    \begin{macrocode}
+  \bool_lazy_or:nnTF
+    { \dim_compare_p:n {\belowdisplayskip  < 0pt } }
+    { \dim_compare_p:n {\belowdisplayshortskip  < 0pt } }
+    {
```

Comment:
I'd go with the `nNn` versions here - they are faster so I typically favour them.

Reply:
Me too, not sure why I didn't there